frontend_lib/src/component/Tags/TagList.jsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import PropTypes from 'prop-types'
+import {
+  sendGlobalFlashMessage,
+  naturalCompare,
+  handleFetchResult
+} from '../../helper.js'
+import classnames from 'classnames'
+import NewTagForm from './NewTagForm.jsx'
+import Tag from './Tag.jsx'
+import {
+  getWorkspaceTagList,
+  getContentTagList,
+  deleteContentTag,
+  putContentTag
+} from '../../action.async.js'
+
+// require('./TagList.styl') // see https://github.com/tracim/tracim/issues/1156
+
+class TagList extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      tagList: []
+    }
+  }
+
+  toggleChecked = tagId => {
+    this.setState(previousState => {
+      const { props } = this
+      const checkedToggled = !previousState.tagIsChecked[tagId]
+      if (checkedToggled) {
+        putContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      } else {
+        deleteContentTag(props.apiUrl, props.workspaceId, props.contentId, tagId)
+      }
+
+      return {
+        tagIsChecked: {
+          ...previousState.tagIsChecked,
+          [tagId]: checkedToggled
+        }
+      }
+    })
+  }
+
+  componentDidMount () {
+    this.updateTagList()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { props } = this
+
+    if (prevProps.contentId !== props.contentId) {
+      this.updateTagList()
+    }
+  }
+
+  async updateTagList () {
+    const { props } = this
```

Comment:
Since this is not an arrow function, shouldn't we bind this this?

Reply:
It is called from methods and not as a callback so this will work without binding.

Reply:
ok, thx, I'll resolve this conversation